### PR TITLE
Ecdc 2729 fix nexus format docs

### PR DESCRIPTION
**Pull request by SkyToGround**
### Description of work

* As the title says.

### CODE REVIEW

**Review by zjttoefs — CHANGES_REQUESTED** [review]
The pull from the upstream repo almost needs no review. Maybe split that out and I'll just merge.

For the changes: Do not delete stuff from the definitions. That is unlikely to get accepted upstream and will be a liability merging. If we do our own version of the definitions - where do we draw the line with the edits anyway?

For the ```depends_on```: The documentation has to include the reference point. If we are unsure about that, make a good guess (required for components we care about), or put in a TODO placeholder.

**Comment by SkyToGround** [comment]
@zjttoefs I have now added reference point documentation to all the types of devices that I feel ok with adding it to.

**Comment by zjttoefs** [comment]
I pushed a few commits and would like some feedback:

- the NXoff_geometry could be used as a tie breaker, when our definition of the reference point is not sufficiently clear or ambiguous due to a complex shape. Does that make sense?
- I added what I think is the correct reference point for a detector
- Made a comment for the chopper reference, which I think helps.
- NXenvironment does not have a depends_on and I would not draw this in our 3d pane, so I reverted the edit

If that shape idea catches on, it could be added in more (if not all) cases. 
I still feel a little lukewarm about editing the clasess where we have no idea what reference location to suggest.

The png's are a very nice touch. Much appreciated @SkyToGround !

File: definitions/base_classes/NXattenuator.nxdl.xml [hunk 1]
```diff
@@ -88,4 +88,30 @@
             for a summary of the discussion.
         </doc>
     </attribute>
+    <field name="depends_on" type="NX_CHAR">
+      <doc>
+        Points to the path to a field defining the last transformation in a chain of transformations
```
Comment (mattclarke) [inline definitions/base_classes/NXattenuator.nxdl.xml:93]:
```suggestion
        Points to the path of a field defining the last transformation in a chain of transformations
```

File: definitions/base_classes/NXattenuator.nxdl.xml [hunk 2]
```diff
@@ -88,4 +88,30 @@
...
+        which defines the position and orientation of this component or the string "." if located
+        in the origin.
+
+        The reference point of the attenuator is its center in the x and y axis. The reference point on the z axis is the
+        surface of the attenuator pointing towards the source.
+
+        For complex geometries, where identifing a centre in x and y is not obvious
```
Comment (mattclarke) [inline definitions/base_classes/NXattenuator.nxdl.xml:100]: centre or center - pick one or the other ;)

Comment (mattclarke) [inline definitions/base_classes/NXattenuator.nxdl.xml:100]:
```suggestion
        For complex geometries, where identifying a centre in x and y is not obvious 
```

File: definitions/base_classes/NXattenuator.nxdl.xml [hunk 3]
```diff
@@ -88,4 +88,30 @@
...
+      </doc>
+    </field>
+    <group type="NXtransformations">
+      <doc>
+        Transformations used by this component to define its position and orientation.
+      </doc>
+    </group>
+    <group name="shape" type="NXoff_geometry">
+      <doc>
+        Shape of by this component. Particulary useful to define the origin for position and orientation in non-standard cases.
```
Comment (mattclarke) [inline definitions/base_classes/NXattenuator.nxdl.xml:114]:
```suggestion
        Shape of this component. Particulary useful to define the origin for position and orientation in non-standard cases.
```

File: definitions/base_classes/NXattenuator.nxdl.xml (hunk 2 above)
Comment (kmurica) [reply definitions/base_classes/NXattenuator.nxdl.xml:100]: I think we belong to the Data Management and Software Centre? :D

File: definitions/base_classes/NXslit.nxdl.xml [hunk 4]
```diff
@@ -41,6 +41,15 @@
 			on "." i.e. the origin of the coordinate system.
 			If desired the location of the slit can also be described relative to
 			an NXbeam, which will allow a simple description of a non-centred slit.
+
+			The reference plane of the slit is orhogonal to the z axis and includes the
```
Comment (ebadkamil) [inline definitions/base_classes/NXslit.nxdl.xml:45]:
```suggestion
			The reference plane of the slit is orthogonal to the z axis and includes the
```

**Comment by ebadkamil** [comment]
LGTM. @zjttoefs must approve.

File: definitions/base_classes/NXdetector.nxdl.xml [hunk 5]
```diff
@@ -817,4 +817,19 @@
...
       </doc>
   </attribute>
+  <field name="depends_on" type="NX_CHAR">
+    <doc>
+      Points to the path to a field defining the last transformation in a chain of transformations
+      which defines the position and orientation of this component or the string "." if located
+      in the origin.
+
+      The reference point of the detector is the center of the first pixel.
+      In complex geometries the NXoff_geometry groups can be used to provide an umabigous reference.
```
Comment (kmurica) [inline definitions/base_classes/NXdetector.nxdl.xml:827]: umabigous -> unambiguous?

File: definitions/utils/nxdl_desc2rst.py [hunk 6]
```diff
@@ -382,7 +382,9 @@ def getDocFromNode(ns, node, retval=None):
             text = lines[0][indent:]
         for line in lines[1:]:
             if not len(line[:indent].strip()) == 0:
```
Comment (kmurica) [inline definitions/utils/nxdl_desc2rst.py:384]: Why not just remove this if statement?

**Review by kmurica — CHANGES_REQUESTED** [review]

File: definitions/base_classes/NXdetector.nxdl.xml (hunk 5 above)
Comment (SkyToGround) [reply definitions/base_classes/NXdetector.nxdl.xml:827]:
```suggestion
      In complex geometries the NXoff_geometry groups can be used to provide an unambiguous reference.
```
There is a beautiful "insert a suggestion" functionality in GitHub. Feel free to use it.

File: definitions/utils/nxdl_desc2rst.py (hunk 6 above)
Comment (SkyToGround) [reply definitions/utils/nxdl_desc2rst.py:384]: To modify the code as little as possible. I do not know what the importance of this piece of code is, only that it is not working.

File: definitions/base_classes/NXattenuator.nxdl.xml (hunk 2 above)
Comment (SkyToGround) [reply definitions/base_classes/NXattenuator.nxdl.xml]: I want to make a note that the original text is inconsistent as well. If I am to change/fix this I would simply do a search and replace on the whole directory.

File: definitions/utils/nxdl_desc2rst.py (hunk 6 above)
Comment (kmurica) [reply definitions/utils/nxdl_desc2rst.py:384]: ok!

**Review by kmurica — APPROVED** [review]
LGTM.

**Comment by kmurica** [comment]
@zjttoefs is this PR in a mergeable state?

**Pull request merged**